packages/admin-client/src/common/components/CustomDatagridRow/CustomDatagridRow.tsx

```diff
@@ -88,7 +92,10 @@ const DatagridRow = React.forwardRef<HTMLTableRowElement, CustomDatagridRowProps
     const handleClick = useCallback(
         async (event) => {
             event.persist()
-            const type = typeof rowClick === "function" ? await rowClick(id as Identifier, resource, record) : rowClick
+            const type: string | boolean =
+                typeof rowClick === "function" && record
+                    ? await rowClick(id as Identifier, resource as string, record)
+                    : (rowClick as string)
```

Comment:
Just curious in which case `type` can be `boolean`?  
Does `await rowClick(id as Identifier, resource as string, record)` return `string | boolean`?

It just a bit confusing to me but please feel free to ignore my comment if there is a logic behind this.

Reply:
Yes, here's the functions type definition: `const rowClick: (id: Identifier, resource: string, record: RaRecord) => string | false | Promise<string | false>`